subprojects/core/src/main/java/org/gradle/internal/build/DefaultBuildLifecycleController.java

```diff
@@ -115,6 +116,20 @@ public void populateWorkGraph(Consumer<? super TaskExecutionGraphInternal> actio
         controller.inState(State.TaskSchedule, () -> workPreparer.populateWorkGraph(gradle, action));
     }
 
+    @Override
+    public void finalizeWorkGraph(boolean workScheduled) {
+        if (workScheduled) {
```

Comment:
I tried to use `controller.inState(State.TaskSchedule)`, though that caused `CompositeBuildTaskLifecycleIntegrationTest.shouldRunAfter task from included build is deprecated` to fail. I don't understand why this doesn't work :(